src/app/games/diam/DiamDummyMinimax.ts

```diff
@@ -0,0 +1,76 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { LegalityStatus } from 'src/app/jscaip/LegalityStatus';
+import { Minimax } from 'src/app/jscaip/Minimax';
+import { NodeUnheritance } from 'src/app/jscaip/NodeUnheritance';
+import { Player } from 'src/app/jscaip/Player';
+import { GameStatus } from 'src/app/jscaip/Rules';
+import { DiamMove, DiamMoveDrop, DiamMoveShift } from './DiamMove';
+import { DiamPiece } from './DiamPiece';
+import { DiamNode, DiamRules } from './DiamRules';
+import { DiamState } from './DiamState';
+
+export class DiamDummyMinimax extends Minimax<DiamMove, DiamState, LegalityStatus> {
+    public getListMoves(node: DiamNode): DiamMove[] {
+        const state: DiamState = node.gameState;
+        const drops: DiamMove[] = this.getListDrops(state);
+        const shifts: DiamMove[] = this.getListShifts(state);
+        return drops.concat(shifts);
+    }
+    private getListShifts(state: DiamState): DiamMoveShift[] {
+        const shifts: DiamMoveShift[] = [];
+        const shiftSources: Coord[] = this.getShiftSources(state);
+        for (const shiftSource of shiftSources) {
+            for (const shift of [new DiamMoveShift(shiftSource, 'clockwise'), new DiamMoveShift(shiftSource, 'counterclockwise')]) {
+                if (DiamRules.get().isLegal(shift, state).legal.isSuccess()) {
+                    shifts.push(shift);
+                }
+            }
+        }
+        return shifts;
+    }
+    private getListDrops(state: DiamState): DiamMoveDrop[] {
+        const remainingPieces: DiamPiece[] = this.getRemainingPiecesForCurrentPlayer(state);
+        const drops: DiamMoveDrop[] = [];
+        for (let x: number = 0; x < DiamState.WIDTH; x++) {
+            for (const piece of remainingPieces) {
+                const drop: DiamMoveDrop = new DiamMoveDrop(x, piece);
+                if (DiamRules.get().isLegal(drop, state).legal.isSuccess()) {
```

Comment:
this.ruler ?

Reply:
hmm c'est pas préférable de passer par les singleton justement, qui sont mieux typés ? (Ici ça change pas grand chose niveau typage, mais dans d'autre minimax oui, où on se retrouve à devoir caster this.ruler)

Reply:
d'ailleurs, suite au léger refactor, c'est soit: `DiamRules.get().dropHeightValidity(drop, state)...`, soit `(this.ruler as DiamRules).dropHeightValidity(drop, state)`